From: Robert Millan <rmh@debian.org>
Subject: Work around gcc bug resulting in build failure

../../src/bsd_mouse.c: In function 'CheckProtocol':
../../src/bsd_mouse.c:142:30: error: array subscript is above array bounds [-Werror=array-bounds]

Debian bug#665390.

Followup-From: Petr Salinger <Petr.Salinger@seznam.cz>
Followup-Bug: #681506

Index: xserver-xorg-input-mouse/src/bsd_mouse.c
===================================================================
--- xserver-xorg-input-mouse.orig/src/bsd_mouse.c
+++ xserver-xorg-input-mouse/src/bsd_mouse.c
@@ -139,10 +139,10 @@ CheckProtocol(const char *protocol)
 {
     int i;
 
-    for (i = 0; internalNames[i]; i++)
+    for (i = -1; internalNames[++i];)
 	if (xf86NameCmp(protocol, internalNames[i]) == 0)
 	    return TRUE;
-    for (i = 0; miscNames[i]; i++)
+    for (i = -1; miscNames[++i];)
 	if (xf86NameCmp(protocol, miscNames[i]) == 0)
 	    return TRUE;
     return FALSE;
